### PR TITLE
boost: fix build on Rosetta

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

A fix for the build on 10.6.x Rosetta. Addresses my own ticket: https://trac.macports.org/ticket/64978

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by barracuda156** [comment]
@mascguy Is this an okay way to add Rosetta case? Alternatively we could just make it for `platform darwin powerpc` (since the latest OS supporting `ppc` is 10.6.8).

File: devel/boost176/Portfile [hunk 1]
```diff
@@ -166,9 +166,16 @@ if {[string match *gcc* ${configure.compiler}]} {
 }
 
 # makecontext/swapcontext introduced in 10.6
-if {${os.platform} eq "darwin" && ${os.major} < 10} {
-    configure.args-append   --without-libraries=context \
-                            --without-libraries=coroutine
+# In 10.6.x Rosetta they do not work: https://trac.macports.org/ticket/64978
```
Comment (michaelld) [inline devel/boost176/Portfile:169]: Prefer that these 2 cases be combined into something the following...
```
if {${os.major} < 10 || {${os.major} == 10 && ${build_arch} eq "ppc"}} {
```

**Comment by michaelld** [comment]
> @mascguy Is this an okay way to add Rosetta case? Alternatively we could just make it for `platform darwin powerpc` (since the latest OS supporting `ppc` is 10.6.8).

I think the issue is with Intel 10.[45] and PPC 10.[456] ... hence using `platform darwin` makes sense, whether as before (within the `if`) or directly

**Review by michaelld — COMMENTED** [review]
I can't check / verify, but I'm OK with this change, or a combined `if` per my note on this review, as it impacts only 10.6 and older, which is a niche group

**Comment by barracuda156** [comment]
@michaelld @mascguy If the present version looks good, I guess it should be added to other Boost versions too.

P. S. I used round brackets to combine conditions, since otherwise I got an error:
```
10:~ svacchanda$ port lint --nitpick boost176
Error: Unable to open port: expected boolean value but got "${os.major} == 10 && ${build_arch} eq "ppc""
10:~ svacchanda$ port lint --nitpick boost176
--->  Verifying Portfile for boost176
--->  0 errors and 0 warnings found.
```

**Review by michaelld — APPROVED** [review]
LGTM ... Good catch on the round versus curly brackets ... I hadn't tried & so went with memory, which clearly wasn't correct! And, yes, we'll want to update other Boost too

**Comment by michaelld** [comment]
@mascguy let's get your review & if you approve then go ahead & merge

**Comment by barracuda156** [comment]
> And, yes, we'll want to update other Boost too

Should I do it with separate commits within this PR or via separate PRs?

**Comment by michaelld** [comment]
> > And, yes, we'll want to update other Boost too
> 
> Should I do it with separate commits within this PR or via separate PRs?

Either will work, but if you want to get it into this PR that would be ideal in my opinion ... keep them all together for the purposes of reviewing / understanding what the change(s) was/were & why

**Comment by barracuda156** [comment]
@mascguy The bots fail to some unrelated reason, it seems. Like here: https://github.com/macports/macports-ports/pull/15564

**Comment by michaelld** [comment]
These changes should not impact the CI/QA, as they target OSX that is not part of the CI/QA here ... not sure what's going on there, but I'm good with merging this given the targeted OSX. @mascguy what do you think?

**Comment by mascguy** [comment]
> These changes should not impact the CI/QA, as they target OSX that is not part of the CI/QA here ... not sure what's going on there, but I'm good with merging this given the targeted OSX. @mascguy what do you think?

I'm good with merging. Thanks guys!

**Review by mascguy — APPROVED** [review]

**Pull request merged**